### ISSUE TITLE
saml2aws 2.0 configuration changes and refactor

### ISSUE THREAD

**Issue by wolfeidau** [Versent/saml2aws#69]
I am currently working on a refactor of the providers, and a new configuration file.

The goals of this release are:

1. Enable configurable MFA per provider, to support #50 
2. Refactor all the providers into their own package to enable a bit more protection, and the flexibility to do more in each.
3. Allow any number of new parameters for providers instead of just skip-verify
4. A general tidy based on feedback from https://goreportcard.com/report/github.com/versent/saml2aws

This work is ongoing in the master branch but given the breaking config change is last this should be OK.

New configuration will be located at `~/.saml2aws`, rather than the current `~/.aws2saml` and look something like:

```ini
[wolfeidau]
hostname = id.wolfe.id.au
username = mark@wolfe.id.au
provider = keycloak
mfa      = totp

[test123]
hostname = id.whatever.com
username = abc@whatever.com
provider = keycloak
mfa      = sms
```

There will be a new configure command, similar to the AWS cli which will enable you to setup IDP accounts and give each a name.

 Outstanding tasks:

- [x] Configuration refactor
- [x] Provider refactor
- [x] Use the provider flag to highlight configuration changes and point to readme.
- [x] Update readme
- [ ] Implement timeouts for http client
- [ ] Debug Logging
- [ ] Pre release build

**Comment by wolfeidau** [comment]
@dan-v I am almost ready to push out a pre release with all the new configuration changes inspired/required by your PR.

Would love your feedback / critique on the new setup once it lands.

Cheers

**Comment by dan-v** [comment]
Awesome! Thanks @wolfeidau !

**Comment by jenwhy-versent** [comment]
That's really cool  @wolfeidau Would be keen to see what the changes are

**Comment by cliff-wakefield** [comment]
Just pulled the latest code and tried 'configure' works a treat.

Anyway to get the profile to update also included in the idp-account info.

Otherwise it seems you still need to do
`~/go/bin/saml2aws --idp-account=somecustomer --profile somecustomer_saml login`

**Comment by wolfeidau** [comment]
@cliff-wakefield Although I don't use the profile option (i just the default one named `saml`) I am interested in some feedback.

Would it be helpful to add the profile you want to use to be saved with the configuration for that idp account as in the example below? 

```
saml2aws configure --profile somecustomer_saml 
```

Rather than prompt for every option I am planning on letting you save flags for all the extra stuff.

**Comment by wolfeidau** [comment]
This is now shipped.

If you have any issues / suggestions open a ticket!

**Comment by cliff-wakefield** [comment]
@wolfeidau yes saving the profile into the saved configuration for the IDP is essential if you want to have multiple AWS CLI sessions active at once which is a requirement for myself.

Now that this request is closed how do you want to handle this additional requirement? Open a new issue? Re-open this one?

**Comment by wolfeidau** [comment]
@cliff-wakefield yes please, I have started working on this already but a ticket would help keep it rolling.

Cheers

**Comment by cliff-wakefield** [comment]
Created issue #102 @wolfeidau
